src/IcedTasks/AsyncEx.fs

```diff
@@ -0,0 +1,381 @@
+    /// </remarks>
+    static member inline AwaitAwaitable(awaitable: 'Awaitable) =
+        AsyncEx.AwaitAwaiter(Awaitable.GetAwaiter awaitable)
+
+    /// <summary>
+    /// Return an asynchronous computation that will wait for the given Task to complete and return
+    /// its result.
+    /// </summary>
+    /// <param name="awaiter">The Awaitable to await</param>
+    ///
+    /// <remarks>
+    /// This is based on <see href="https://github.com/fsharp/fslang-suggestions/issues/840">Async.Await overload (esp. AwaitTask without throwing AggregateException)</see>
+    /// </remarks>
+    static member AwaitTask(task: Task) : Async<unit> =
+        Async.FromContinuations(fun (onNext, onError, onCancel) ->
+            if task.IsCompleted then
+                if task.IsFaulted then
+                    let e = task.Exception
+
+                    if e.InnerExceptions.Count = 1 then
+                        onError e.InnerExceptions.[0]
+                    else
+                        onError e
+                elif task.IsCanceled then
+                    onCancel (TaskCanceledException(task))
+                else
+                    onNext ()
+            else
+                task.ContinueWith(
+                    (fun (task: Task) ->
+                        if task.IsFaulted then
+                            let e = task.Exception
+
+                            if e.InnerExceptions.Count = 1 then
+                                onError e.InnerExceptions.[0]
+                            else
+                                onError e
+                        elif task.IsCanceled then
+                            onCancel (TaskCanceledException(task))
+                        else
+                            onNext ()
+                    ),
+                    TaskContinuationOptions.ExecuteSynchronously
+                )
+                |> ignore
+        )
+
+    /// <summary>
+    /// Return an asynchronous computation that will wait for the given Task to complete and return
+    /// its result.
+    /// </summary>
+    /// <param name="awaiter">The Awaitable to await</param>
+    ///
+    /// <remarks>
+    /// This is based on <see href="https://github.com/fsharp/fslang-suggestions/issues/840">Async.Await overload (esp. AwaitTask without throwing AggregateException)</see>
+    /// </remarks>
+    static member AwaitTask(task: Task<'T>) : Async<'T> =
+        Async.FromContinuations(fun (onNext, onError, onCancel) ->
+
+            if task.IsCompleted then
+                if task.IsFaulted then
+                    let e = task.Exception
+
+                    if e.InnerExceptions.Count = 1 then
+                        onError e.InnerExceptions.[0]
+                    else
+                        onError e
+                elif task.IsCanceled then
```

Comment:
@TheAngryByrd the official ATC calls onError, and does not wrapp the inner one
IIRC this has an important impact (the fssnip has tests that expose it). I cant recall whether the bit that breaks this way is
1. Async.Parallel doesn't kill the siblings (as it;'s waiting for the first exception or something?)
2. cancellation is not honored

IIRC there are two separate F# issues that cover the  overall requirements (the AggregateException bit vs the Await Cancellation as Exception aspect)

My general conclusion is that we're better off having AwaitTaskCorrect, which has been very well battle proven, with specific failing tests covering any deviation from its entire behavior